File: guests/swift/window.swift
// The window scene, Swift port — guests/rust/window.rs,
// tools/scenes/window.steps.

import Foundation

let app = KayaApp()

app.build { tx in
    tx.window(title: "window probe", width: 640.0, height: 400.0)
    let probe = tx.signal(.str("window probe"))
    let root = tx.column {
        tx.label(bind: probe)  // label#0
    }
    tx.mount(root)
}

app.run()

File: guests/ocaml/window.ml
(* The window scene, OCaml port — guests/rust/window.rs,
   tools/scenes/window.steps. *)

open Kaya_wire
open Kaya_app

let () =
  let app = Kaya_app.create () in

  build app (fun () ->
     window ~title:"window probe" ~width:640.0 ~height:400.0 ();
     let probe = signal (Str "window probe") in

     let root = column [ label ~bind:probe (* label#0 *) ] () in
     mount root);

  exit (run app)
